apps/minifront/src/state/status.ts

```diff
@@ -1,36 +1,37 @@
-import { SliceCreator } from '.';
-import { viewClient } from '../clients';
+import { ZQueryState } from '@penumbra-zone/zquery/src/types';
+import { SliceCreator, useStore } from '.';
+import { createZQuery } from '@penumbra-zone/zquery';
+import { getStatusStream } from '../fetchers/status';
 
-export interface StatusSlice {
-  start: () => Promise<void>;
+interface Status {
   fullSyncHeight?: bigint;
   latestKnownBlockHeight?: bigint;
-  error?: unknown;
 }
 
-export const createStatusSlice = (): SliceCreator<StatusSlice> => set => ({
-  start: async () => {
-    try {
-      // statusStream sends new data to stream only when a new block is detected.
-      // This can take up to 5 seconds (time of new block generated).
-      // Therefore, we need to do a unary request to start us off.
-      const status = await viewClient.status({});
-      set(state => {
-        state.status.fullSyncHeight = status.fullSyncHeight;
-        if (!status.catchingUp) state.status.latestKnownBlockHeight = status.fullSyncHeight;
-      });
-
-      const stream = viewClient.statusStream({});
-      for await (const result of stream) {
-        set(state => {
-          state.status.fullSyncHeight = result.fullSyncHeight;
-          state.status.latestKnownBlockHeight = result.latestKnownBlockHeight;
-        });
-      }
-    } catch (error) {
-      set(state => {
-        state.status.error = error;
-      });
-    }
+export const { status, useStatus } = createZQuery({
+  name: 'status',
+  fetch: getStatusStream,
+  stream: (_prevState: Status | undefined, item: Status): Status => ({
+    fullSyncHeight: item.fullSyncHeight,
+    latestKnownBlockHeight: item.latestKnownBlockHeight,
+  }),
+  getUseStore: () => useStore,
+  get: state => state.status.status,
+  set: setter => {
+    const newState = setter(useStore.getState().status.status);
+    useStore.setState(state => {
+      state.status.status = newState;
+    });
```

Comment:
[question](https://conventionalcomments.org/): from the DX perspective, can `setter` accept a `newState` argument with the `getter()` return, so that we won't have to query it every time?

Reply:
Good call. I'd originally used the design you described, then had to switch to this design for some TypeScript reason. I think that reason has since become moot, so I should be able to switch it back. Will look into that and follow up. Thanks for catching that! (This DX is indeed quite confusing.)

Reply:
OK, wasn't able to resolve this. For other reviewers' benefit, I've messaged Max separately about pairing on resolving the TS issues related to this; until then, though, I'd still love a PR review on this!